ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -2105,7 +2105,13 @@ private void getMetaData(QB qb, ReadEntity parentInput)
       Table tab = getTableObjectByName(tabName, false);
       if (tab != null) {
         // do a deep copy, in case downstream changes it.
```

Comment:
Can we update the comment since this is no longer deep copy?

Reply:
This has not changed, it is still creating a deep copy of the metastore table.
```
...tab.getTTable().deepCopy()...
```

Reply:
The constraints are not deep copied though, you are right. Those should not change in any case; I will change the comment accordingly.